core/codegen/src/derive/form_field.rs

```diff
@@ -291,5 +292,39 @@ pub fn validators<'v>(
             })).unwrap()
         });
 
-        Ok(exprs)
+    Ok(exprs)
+}
+
+pub fn default<'v>(
+    field: Field<'v>,
+) -> Result<Option<syn::Expr>> {
+    let mut exprs = FieldAttr::from_attrs(FieldAttr::NAME, &field.attrs)?
+        .into_iter()
+        .map(|a| a.default)
+        .flat_map(move |expr| {
+            let field_span = field.ident().span()
+                .join(field.ty.span())
+                .unwrap_or(field.ty.span());
+
+            expr.map(|expr| {
+                let span = expr.key_span.unwrap_or(field_span);
+                define_spanned_export!(span => _form);
+                syn::parse2(quote_spanned!(span => {
+                    let __default: Option<_> = { #expr }.into();
+                    __default
+                })).unwrap()
+            })
+        });
+    let first = exprs.next();
+    if first.is_none() {
+        return Ok(None);
+    }
+    let second: Option<syn::Expr> = exprs.next();
+    if let Some(expr) = second {
+        return Err(expr.span()
+            .error("unexpected second default")
+            .note("only one default is allowed in this context"));
+    }
+
+    Ok(first)
 }
```

Comment:
```suggestion
pub fn default<'v>(
    field: Field<'v>,
) -> Result<Option<syn::Expr>> {
    let mut exprs = FieldAttr::from_attrs(FieldAttr::NAME, &field.attrs)?
        .into_iter()
        .filter_map(|a| a.default)
        .map(move |expr| {
            let ty = &field.ty;
            define_spanned_export!(expr.span => _form);
            syn::parse2(quote_spanned!(expr.span => {
                let __default: Option<#ty> = { #expr }.into();
                __default
            })).unwrap()
        });

    let first: Option<syn::Expr> = exprs.next();
    if let Some(expr) = exprs.next() {
        return Err(expr.span()
            .error("duplicate `default` form field attribute")
            .help("form fields may have at-most one `default`"));
    }

    Ok(first)
}
```

A few notes:

  1. Use `filter_map` to avoid the `flat_map`.
  2. No need to check if `first` is `None` and return `None`; just return it.
  3. We want to use the span of the `expr` as the user wrote.
  4. Use the known type in `Option<#ty>`, otherwise inference will surely fail in some cases.